jupyter_client/jsonutil.py

```diff
@@ -75,7 +83,8 @@ def squash_dates(obj):
 def date_default(obj):
     """default function for packing datetime objects in JSON."""
     if isinstance(obj, datetime):
-        return obj.isoformat()
+        obj = _ensure_tzinfo(obj)
```

Comment:
Fair enough. Then in some future version, we could potentially drop support for passing naive datetime objects to the serialisation machinery.